File: problems/python/1-arrays-hashing/0028_str_str/solution.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        if needle == "":
            return -1
        for i in range(0, len(haystack)):
            if needle[0] == haystack[i] and haystack[i:(i+len(needle))] == needle:
                return i
        return -1

File: problems/typescript/1-arrays-hashing/0028_str_str/solution.ts
/**
 * Find the Index of the First Occurrence in a String (strStr)
 * Time: O(n*m), Space: O(m)
 *
 * Manual substring matching
 * Note: substring() creates a new string of length m
 */
export function strStr(haystack: string, needle: string): number {
	for (let i = 0; i < haystack.length; i++) {
		if (
			haystack[i] === needle[0] &&
			haystack.substring(i, i + needle.length) === needle
		) {
			return i;
		}
	}
	return -1;
}
